python/cudf/cudf/_lib/transform.pyx

```diff
@@ -165,3 +165,23 @@ def table_encode(Table input):
         ),
         Column.from_unique_ptr(move(c_result.second))
     )
+
+
+def one_hot_encode(Column input_column, Column categories):
+    cdef column_view c_view_input = input_column.view()
+    cdef column_view c_view_categories = categories.view()
+    cdef pair[unique_ptr[column], table_view] c_result
+
+    with nogil:
+        c_result = move(
+            libcudf_transform.one_hot_encode(c_view_input, c_view_categories)
+        )
+
+    owner = Column.from_unique_ptr(move(c_result.first))
+    encodings, _ = data_from_table_view(
+        move(c_result.second),
+        owner=owner,
+        column_names=range(categories.size)
```

Comment:
Should the column names be the values of the categories instead?

Reply:
I didn't do that because
- the current one_hot_encoding interface doesn't seem to require returning the categories but just the list of the columns. and
- in `get_dummies` the columns names are added as a post processing step.

Likely this suggestion can be included when we deprecate the `one_hot_encoding` interface and implement `get_dummies` directly with the cython interface. For now, it seems redundant here - and because `column_names` are iterables, it seems least costly to use a range here.